ethcore/src/blockchain/blockchain.rs

```diff
@@ -430,7 +430,10 @@ impl<'a> Iterator for EpochTransitionIter<'a> {
 				return None
 			}
 
-			let transitions: EpochTransitions = ::rlp::decode(&val[..]);
+			let transitions: EpochTransitions = match ::rlp::decode(&val[..]) {
+				Ok(decoded) => decoded,
+				Err(_) => return None
+			};
```

Comment:
Shouldn't break out of the iterator with invalid RLP. Since we wrote the `EpochTransitions` to the db in the first place then either the db is corrupted or the data structure has changed, so should panic.